### PR TITLE
Created SemanticLogger and associated Console and Mongo loggers

### PR DESCRIPTION
**Pull request by elimisteve**
Also created Payload data type for storing logging information, getting as close to the Ruby semantic_logger as possible.

### CODE REVIEW

File: semantic_logger.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+// Define methods necessary for MultiSemanticLogger to implement
+// SemanticLogger
+//
+
+func (ml MultiSemanticLogger) Log(level LogLevel, msg string, payload *Payload,
+	excep *Exception) error {
+	switch level {
+	case TRACE, DEBUG, INFO, WARN, ERROR, FATAL:
+		for _, lgr := range ml {
+			// TODO: Decide what to do with returned `error` value
```
Comment (erikstmartin) [inline semantic_logger.go]: writing it to stdout is probably a safe bet

**Comment by erikstmartin** [comment]
We'll definitely need a FileLogger, at some point someone could do a syslog one if they wanted as well

File: semantic_logger.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	"log"
+	"runtime"
+	"strings"
+	"time"
+)
+
+type Exception struct {
+	Message   string   `json:"message"`
+	Backtrace []string `json:"backtrace"`
+}
```
Comment (erikstmartin) [inline semantic_logger.go:35]: We should really provide a way to help accomodate filling this backtrace, for example something like: https://gist.github.com/3899233

Comment (skelterjohn) [inline semantic_logger.go:35]: You can also use something like this: http://code.google.com/p/gonicetrace/ (I am the author, feel free to copy/paste code without citation if you want any of it)

File: semantic_logger.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	"fmt"
+	"io"
+	"labix.org/v2/mgo"
+	"log"
+	"runtime"
+	"strings"
+	"time"
+)
+
+type Exception struct {
```
Comment (skelterjohn) [inline semantic_logger.go]: why do we need to consider exceptions at all?

File: semantic_logger.go (hunk 1 above)
Comment (skelterjohn) [inline semantic_logger.go]: what returned error value?

File: semantic_logger.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+func (excep *Exception) String() string {
+	// message << " -- " << "#{exception.class}: #{exception.message}\n
+	// #{(exception.backtrace || []).join("\n")}"
+	formatStr := "%s -- %s: %s\n%s"
+	backtrace := strings.Join(excep.Backtrace, "\n")
+	return fmt.Sprintf(formatStr, excep.Message, "panic", excep.Message,
+		backtrace)
+}
+
+type Payload struct {
```
Comment (skelterjohn) [inline semantic_logger.go]: this type should have godoc some comments

File: semantic_logger.go (hunk 1 above)
Comment (elimisteve) [inline semantic_logger.go]: @skelterjohn The `Log` method returns an `error`

File: semantic_logger.go (hunk 4 above)
Comment (elimisteve) [inline semantic_logger.go]: Will add

File: semantic_logger.go (hunk 3 above)
Comment (elimisteve) [inline semantic_logger.go]: Two comments:
1. The idea is that within a `recover()` block we'd store the stacktrace info from panics. I'm calling them "exceptions" to maintain compatibility with [semantic_logger](https://github.com/ClarityServices/semantic_logger#semantic_logger).
2. The `Exception` type is removed in later commits; this info should be part of the `Payload` struct.

File: semantic_logger.go (hunk 1 above)
Comment (elimisteve) [inline semantic_logger.go]: @erikstmartin That's what it does in the latest version, which was also part of the Pull Request

File: semantic_logger.go (hunk 2 above)
Comment (elimisteve) [inline semantic_logger.go:35]: @erikstmartin This code, too, is included in the Pull Request. It's based upon some very helpful code that John wrote for the original logger.

**Pull request merged**